Provide parameter for rnastar index builder to increase available RAM

lparsons opened the pull request:
This is required if building large or complex genomes.

yhoogstrate commented on data_managers/data_manager_star_index_builder/data_manager/rna_star_index_builder.xml:
```diff
@@ -60,6 +61,8 @@
             <when value="withoutGTF" />
         </conditional>
+        <param argument="--limitGenomeGenerateRAM" type="integer" min="1000000" value="31000000000" label="Maximum available RAM (bytes) for genome generation"
+            help="For custom genomes, the rule of thub is to multiply the size of the reference FASTA file by 9 to estimated required amount of RAM." />
```
thumb

yhoogstrate commented:
Would it make sense to determine the size of the fasta file(s) (`du -s`, or in Cheetah, for example when there are `fa.gz` files) and multiply it by 9, and then chech if this does not exceed `\$GALAXY_MEMORY_MB`?

lparsons commented:
@yhoogstrate I don't think that will help me to generate references that require >31GB of memory (which is my goal). One thought:
1. Use [dynamic job destinations](https://docs.galaxyproject.org/en/master/admin/jobs.html#dynamic-destination-mapping-dtd-method) to increase the memory limit for the job as required.
2. Use `\$GALAXY_MEMORY_MB` in Cheetah to set the `--limitGenomeGenerateRAM` parameter.

However, I don't really know how to do that properly. Two questions:
1. How can I reliably determine the size of the FASTA reference (*i.e.* how do I handle `fa.gz` files) and can I somehow do math (mult by 9) in a DTD or is this going to require [writing python](https://docs.galaxyproject.org/en/master/admin/jobs.html#dynamic-destination-mapping-python-method)? I'd really rather not get that complicated if I can avoid it for what amounts to a simple requirement.
2. Will `\$GALAXY_MEMORY_MB` be set properly based on the DTD rules? How is that variable actually determined?

bgruening commented:
@lparsons by default this tool is using all the memory available right? Why can you not set the amount of memory for this tool in your job_conf.xml file? This is how we do it and we never had problems so far.

lparsons commented:
Thanks for the input everyone. I've moved over to using `\${GALAXY_MEMORY_MB}` (after converting it to bytes). I also added a few other advanced parameters that should help, esp. in some special cases I have (many contigs and small genomes).

I'd appreciate any tips on how I can most reliably get the size of the FASTA file in bytes using [dynamic job destinations](https://docs.galaxyproject.org/en/master/admin/jobs.html#dynamic-destination-mapping-dtd-method).

nsoranzo commented:
> I'd appreciate any tips on how I can most reliably get the size of the FASTA file in bytes using dynamic job destinations.

In bash, something like: `grep -v '^>' decompressed_fasta |wc -m` (zgrep/bzgrep if compressed).

lparsons commented:
> > I'd appreciate any tips on how I can most reliably get the size of the FASTA file in bytes using dynamic job destinations.
>
> In bash, something like: `grep -v '^>' decompressed_fasta |wc -m` (zgrep/bzgrep if compressed).

That would certainly work in bash, but I was more interested in how to do this using the YAML-based DSL ([`config/tool_destinations.yml.sample`](https://github.com/galaxyproject/galaxy/blob/dev/config/tool_destinations.yml.sample)). I don't think anything in the tool XML here will work, since I need to change the SLURM params to increase memory.

bgruening commented:
Thanks @lparsons!
